cypress/e2e/device.cy.ts

```diff
@@ -0,0 +1,74 @@
+import Device from '../../lib/twilio/device';
+import Call from '../../lib/twilio/call';
+import { generateAccessToken } from '../../tests/lib/token';
+
+describe('Device', function () {
+  this.timeout(10000);
+
+  let device1: Device;
+  let device2: Device;
+  let identity1: string;
+  let identity2: string;
+  let token1: string;
+  let token2: string;
+
+  before(() => {
+    identity1 = 'id1-' + Date.now();
+    identity2 = 'id2-' + Date.now();
+    token1 = generateAccessToken(identity1);
+    token2 = generateAccessToken(identity2);
+
+    device1 = new Device(token1);
+    device2 = new Device(token2);
+
+    return Promise.all([device1.register(), device2.register()]);
+  });
+
+  after(() => {
+    if (device1) {
+      device1.disconnectAll();
+      device1.destroy();
+    }
+
+    if (device2) {
+      device2.disconnectAll();
+      device2.destroy();
+    }
+  });
+
+  describe('device 1 calls device 2', () => {
+    let call1: Call;
+    let call2: Call;
+
+    before(
+      () =>
+        new Promise(async (resolve) => {
```

Comment:
Recommend leaving the `Promise<void>`. I believe that more modern versions of Typescript will throw without defining the generic. (Not sure why it's actually building here tbh, we did update to a modern Typescript version lol.)